packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -145,47 +100,25 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     /// @param _gasLimit          Initial gas limit.
     /// @param _unsafeBlockSigner Initial unsafe block signer address.
     /// @param _config            Initial ResourceConfig.
-    /// @param _startBlock        Starting block for the op-node to search for logs from.
-    ///                           Contracts that were deployed before this field existed
-    ///                           need to have this field set manually via an override.
-    ///                           Newly deployed contracts should set this value to uint256(0).
-    /// @param _batchInbox        Batch inbox address. An identifier for the op-node to find
-    ///                           canonical data.
-    /// @param _addresses         Set of L1 contract addresses. These should be the proxies.
     function initialize(
         address _owner,
         uint256 _overhead,
         uint256 _scalar,
         bytes32 _batcherHash,
         uint64 _gasLimit,
         address _unsafeBlockSigner,
-        ResourceMetering.ResourceConfig memory _config,
-        uint256 _startBlock,
-        address _batchInbox,
-        SystemConfig.Addresses memory _addresses
+        ResourceMetering.ResourceConfig memory _config
     )
         public
-        reinitializer(Constants.INITIALIZER)
+        initializer
     {
         __Ownable_init();
         transferOwnership(_owner);
-
-        // These are set in ascending order of their UpdateTypes.
-        _setBatcherHash(_batcherHash);
```

Comment:
We should keep the setters here ie emitting the events in the call to `initialize`. It reduces the amount of information the `op-node` needs for genesis

Reply:
OK will do.

Reply:
I did this, and otherwise significantly reduced the diff in SystemConfig. Thanks for calling that out.